src/llm/apis/openai_completions.cpp

```diff
@@ -224,30 +237,58 @@ absl::Status OpenAIChatCompletionsHandler::parseMessages() {
                             std::size_t pos = url.find(pattern);
                             std::string decoded;
                             if (pos != std::string::npos) {
+                                SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Loading image from base64 string");
                                 size_t offset = pos + pattern.length();
                                 if (!absl::Base64Unescape(std::string_view(url.data() + offset, url.size() - offset), &decoded)) {
                                     return absl::InvalidArgumentError("Invalid base64 string in request");
                                 }
+                                try {
+                                    ov::Tensor tensor = loadImageStbiFromMemory(decoded);
+                                    request.imageHistory.push_back({i, tensor});
+                                } catch (std::runtime_error& e) {
+                                    std::stringstream ss;
+                                    ss << "Image parsing failed: " << e.what();
+                                    SPDLOG_LOGGER_ERROR(llm_calculator_logger, ss.str());
+                                    return absl::InvalidArgumentError(ss.str());
+                                }
                             } else if (std::regex_match(url.c_str(), std::regex("^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$"))) {
+                                SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Loading image using curl");
                                 curl_global_init(CURL_GLOBAL_ALL);
                                 int64_t sizeLimit = 20000000;  // restrict single image size to 20MB
```

Comment:
unrelated question - do we have such restriction in general, also for other loading paths or is it just for using curl?